File: Estudiante.java
import java.util.ArrayList;

public class Estudiante {

    // Atributos privados (encapsulación)
    private String nombre;
    private int edad;
    private String carrera;
    private ArrayList<String> materias;

    // Constructor
    public Estudiante(String nombre, int edad, String carrera) {
        this.nombre = nombre;
        this.edad = edad;
        this.carrera = carrera;
        this.materias = new ArrayList<>();
    }

    // Método para inscribir materia
    public void inscribirMateria(String materia) {
        this.materias.add(materia);
        System.out.println(nombre + " se inscribió en " + materia);
    }

    // Método para presentarse
    public void presentarse() {
        System.out.println("Hola, soy " + nombre);
        System.out.println("Tengo " + edad + " años");
        System.out.println("Estudio " + carrera);
        System.out.println("Materias inscritas: " + materias.size());
    }

    // NUEVO método estudiar
    public void estudiar(int horas) {
        System.out.println(nombre + " estudió " + horas + " horas");
    }

    // Método main para probar
    public static void main(String[] args) {

        // Crear objetos del ejemplo
        Estudiante est1 = new Estudiante("Ana García", 20, "Ing. Sistemas");
        Estudiante est2 = new Estudiante("Carlos López", 22, "Ing. Sistemas");

        // Crear TU estudiante
        Estudiante est3 = new Estudiante(
            "Juan Sebastian Ramos Ovalle",
            17,
            "Sistematización de Datos"
        );

        est3.presentarse();

        System.out.println("---");

        // Inscribir tus 3 materias
        est3.inscribirMateria("Física Newtoniana");
        est3.inscribirMateria("Cálculo Integral");
        est3.inscribirMateria("Estructura de Datos");

        System.out.println("---");

        // Usar el método estudiar
        est3.estudiar(4);
    }
}

File: estudiante.py
class Estudiante:
    # Constructor - se ejecuta al crear un objeto
    def __init__(self, nombre, edad, carrera):
        self.nombre = nombre
        self.edad = edad
        self.carrera = carrera
        self.materias = []

    # Método para agregar materias
    def inscribir_materia(self, materia):
        self.materias.append(materia)
        print(f'{self.nombre} se inscribió en {materia}')

    # Método para mostrar información
    def presentarse(self):
        print(f'Hola, soy {self.nombre}')
        print(f'Tengo {self.edad} años')
        print(f'Estudio {self.carrera}')
        print(f'Materias inscritas: {len(self.materias)}')


# Crear objetos (instancias de la clase)
estudiante1 = Estudiante('Ana García', 20, 'Ingeniería de Sistemas')
estudiante2 = Estudiante('Carlos López', 22, 'Ingeniería de Sistemas')

# Usar los métodos
estudiante1.presentarse()
print('---')
estudiante1.inscribir_materia('POO')
estudiante1.inscribir_materia('Bases de Datos')

print('---')
estudiante2.presentarse()
estudiante2.inscribir_materia('POO')
